### DISCUSSION TITLE
How to convert a string to integer in berry ?

### DISCUSSION THREAD

**Discussion by ouinouin** [arendst/Tasmota#16946] [Q&A] [answered]
Hi, 
first i know there is a better way to proceed in my specific case, but  i d like to know how to convert a string to an integer or a float in berry. there is no implicit conversion as shows this error :-) , i ve tried to see in bytes but i did get nowhere.

any hints  ? 
```

print(mystring)
604
mystring +3
BRY: Exception> 'type_error' - unsupported operand type(s) for +: 'string' and 'int'
stack traceback:
	<unknown source>: in function `main`

```

**Comment by ouinouin** [comment, marked as answer]
i m auto answering, the function to do that is number()
https://github.com/berry-lang/berry/wiki/Chapter-7#number-function

**Reply by s-hadinger** [reply]
Or simply `int()` if you want an integer and not a float

**Reply by ouinouin** [reply]
ok , can also be. @s-hadinger  out of subject but while i find you here :-) 
while navigating the doc (its difficult to do a find in the github wiki) i noticed the "to upper" of the string library i ve seen in some tasmota examples seems not documented.
